vault/expiration.go

```diff
@@ -900,18 +900,18 @@ func (m *ExpirationManager) Register(req *logical.Request, resp *logical.Respons
 		// want to revoke a generated secret (since an error means we may not
 		// be successfully tracking it), remove indexes, and delete the entry.
 		if retErr != nil {
-			revResp, err := m.router.Route(m.quitContext, logical.RevokeRequest(req.Path, resp.Secret, resp.Data))
+			revResp, err := m.router.Route(ctx, logical.RevokeRequest(req.Path, resp.Secret, resp.Data))
```

Comment:
Since in this case we really do want to ensure revocation of any generated secret, I think m.quitContext is correct here.